pkg/controller/build/build_controller.go

```diff
@@ -540,6 +590,47 @@ func (ctrl *Controller) syncMachineConfigPool(key string) error {
 	return ctrl.syncAvailableStatus(pool)
 }
 
+func (ctrl *Controller) markBuildInterrupted(ps *poolState) error {
+	klog.Errorf("Build interrupted for pool %s", ps.Name())
+
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		mcp, err := ctrl.mcfgclient.MachineconfigurationV1().MachineConfigPools().Get(context.TODO(), ps.Name(), metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		ps := newPoolState(mcp)
+		ps.SetBuildConditions([]mcfgv1.MachineConfigPoolCondition{
+			{
+				Type:   mcfgv1.MachineConfigPoolBuildInterrupted,
+				Reason: "BuildInterrupted",
```

Comment:
question: Resuming from an interrupted build is out-of-scope for this PR, correct?

Reply:
correct. I think this might fall into the Rebuild card you have assigned to me though, so laying the groundwork won't hurt.